central/complianceoperator/v2/scanconfigurations/datastore/datastore_impl.go

```diff
@@ -354,6 +357,67 @@ func (d *datastoreImpl) GetProfilesNames(ctx context.Context, q *v1.Query) ([]st
 	return profileNames, err
 }
 
+// GetProfilesNames gets the list of all distinct profile names which were configured either via StackRox or directly
+// in Compliance Operator for the query.
+func (d *datastoreImpl) GetProfilesNames(ctx context.Context, q *v1.Query) ([]string, error) {
+	var err error
+	q, err = withSACFilter(ctx, resources.Compliance, q)
+	if err != nil {
+		return nil, err
+	}
+
+	// get all profiles configured in StackRox compliance schedules
+	resultProfileNames, err := d.getScanConfigProfileNames(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+
+	// get all profiles configured via Compliance Operator, not managed by StackRox
+	scanProfileNames, err := d.GetProfilesNamesFromOperatorScanSettings(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+
+	// merge profile name lists
+	for _, scanProfileName := range scanProfileNames {
+		if !slices.Contains[[]string, string](resultProfileNames, scanProfileName) {
+			resultProfileNames = append(resultProfileNames, scanProfileName)
+		}
+	}
+
+	return resultProfileNames, err
+}
+
+func (d *datastoreImpl) getScanConfigProfileNames(ctx context.Context, q *v1.Query) ([]string, error) {
+	clonedQuery := q.Clone()
+	// Build the select and group by on distinct profile name
+	clonedQuery.Selects = []*v1.QuerySelect{
+		search.NewQuerySelect(search.ComplianceOperatorConfigProfileName).Distinct().Proto(),
```

Comment:
The `ComplianceOperatorScanConfigurationV2Schema` table links to a profile by its name, while the `ComplianceOperatorScanV2Schema` uses the `profileRefId`.
I couldn't figure out way to combine the query for both. I think we need a foreign key to profiles in `ComplianceOperatorScanConfigurationV2Schema` or implement the pagination in the app-layer.

I am not sure at this point which approach is better and how our search framework could support a query like this.